lib/Containers/MerkleTree.cpp

```diff
@@ -474,16 +461,67 @@ std::unique_ptr<MerkleTree<BranchingBits, LockStripes>> MerkleTree<BranchingBits
       new MerkleTree<BranchingBits, LockStripes>(*this));
 }
 
+template <std::size_t const BranchingBits, std::size_t const LockStripes>
+std::unique_ptr<MerkleTree<BranchingBits, LockStripes>>
+MerkleTree<BranchingBits, LockStripes>::cloneWithDepth(std::size_t newDepth) {
```

Comment:
It looks like this method is not modifying the original tree (good!), so we could make it `const`.